### PR TITLE
Publish snapshots to maven via GHA, fix spotless errors, and bump bwc version

### PR DESCRIPTION
**Pull request by kaituo**
### Description
We are de-coupling the task of publishing the maven snapshots from centralized build workflow to individual repositories. This PR publishes maven snapshots using GitHub Actions.

Testing done:
1. ran ./gradlew publishPluginZipPublicationToSnapshotsRepository according to https://github.com/opensearch-project/anomaly-detection/issues/813

Output:
```
no changes added to commit (use "git add" and/or "git commit -a")
(base) kaituo@88665a53bc93 anomaly-detection % find snapshots | sort
snapshots
snapshots/org
snapshots/org/opensearch
snapshots/org/opensearch/plugin
snapshots/org/opensearch/plugin/opensearch-anomaly-detection
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/maven-metadata.xml
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/maven-metadata.xml.md5
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/maven-metadata.xml.sha1
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/maven-metadata.xml.sha256
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/maven-metadata.xml.sha512
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/opensearch-anomaly-detection-3.0.0.0-20230301.220801-1.pom
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/opensearch-anomaly-detection-3.0.0.0-20230301.220801-1.pom.md5
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/opensearch-anomaly-detection-3.0.0.0-20230301.220801-1.pom.sha1
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/opensearch-anomaly-detection-3.0.0.0-20230301.220801-1.pom.sha256
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/opensearch-anomaly-detection-3.0.0.0-20230301.220801-1.pom.sha512
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/opensearch-anomaly-detection-3.0.0.0-20230301.220801-1.zip
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/opensearch-anomaly-detection-3.0.0.0-20230301.220801-1.zip.md5
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/opensearch-anomaly-detection-3.0.0.0-20230301.220801-1.zip.sha1
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/opensearch-anomaly-detection-3.0.0.0-20230301.220801-1.zip.sha256
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/3.0.0.0-SNAPSHOT/opensearch-anomaly-detection-3.0.0.0-20230301.220801-1.zip.sha512
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/maven-metadata.xml
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/maven-metadata.xml.md5
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/maven-metadata.xml.sha1
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/maven-metadata.xml.sha256
snapshots/org/opensearch/plugin/opensearch-anomaly-detection/maven-metadata.xml.sha512
```
### Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/813

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/maven-publish.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Publish snapshots to maven
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [
+        main
+          1.*
+          2.*
```
Comment (gaiksaya) [inline .github/workflows/maven-publish.yml:9]: Looks like this format has caused issues in other repos.
Can you make it 
```
- 'main'
- '1.x'
- '2.x'
```

**Review by gaiksaya — APPROVED** [review]
Added `PUBLISH_SNAPSHOTS_ROLE` role to the secrets

**Review by amitgalitz — DISMISSED** [review]

**Comment by kaituo** [comment]
compile failure due to previous commits:

src\main\java\org\opensearch\ad\AnomalyDetectorPlugin.java
          @@ -175,9 +175,9 @@
           import�org.opensearch.common.settings.SettingsFilter;
           import�org.opensearch.common.unit.TimeValue;
           import�org.opensearch.common.util.concurrent.OpenSearchExecutors;
          +import�org.opensearch.common.xcontent.XContentParserUtils;
           import�org.opensearch.core.xcontent.NamedXContentRegistry;
           import�org.opensearch.core.xcontent.XContentParser;
          -import�org.opensearch.common.xcontent.XContentParserUtils;
           import�org.opensearch.env.Environment;
           import�org.opensearch.env.NodeEnvironment;
           import�org.opensearch.jobscheduler.spi.JobSchedulerExtension;
      src\main\java\org\opensearch\ad\AnomalyDetectorProfileRunner.java
          @@ -61,9 +61,9 @@
           import�org.opensearch.client.Client;
           import�org.opensearch.cluster.node.DiscoveryNode;
           import�org.opensearch.common.xcontent.LoggingDeprecationHandler;
          +import�org.opensearch.common.xcontent.XContentType;
           import�org.opensearch.core.xcontent.NamedXContentRegistry;
           import�org.opensearch.core.xcontent.XContentParser;
          -import�org.opensearch.common.xcontent.XContentType;
           import�org.opensearch.search.SearchHits;
           import�org.opensearch.search.aggregations.Aggregation;
           import�org.opensearch.search.aggregations.AggregationBuilder;
      src\main\java\org\opensearch\ad\cluster\ADDataMigrator.java
          @@ -51,8 +51,8 @@
           import�org.opensearch.ad.util.ExceptionUtil;
           import�org.opensearch.client.Client;
           import�org.opensearch.cluster.service.ClusterService;
          +import�org.opensearch.common.xcontent.XContentFactory;
           import�org.opensearch.core.xcontent.NamedXContentRegistry;
          -import�org.opensearch.common.xcontent.XContentFactory;
           import�org.opensearch.core.xcontent.XContentParser;
      ... (2 more lines that didn't fit)

**Review by amitgalitz — APPROVED** [review]

**Review by gaiksaya — APPROVED** [review]
Please update the PR title and description to reflect additional changes in this PR.
Thanks!

**Comment by kaituo** [comment]
> Please update the PR title and description to reflect additional changes in this PR. Thanks!

updated.

**Pull request merged**